File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findDuplicate = function(nums) {
  let unique=new Set();
  for (let num of nums){
  if(unique.has(num)){
    return num;
  }
  unique.add(num);
  }
};

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int n=nums.size();
        vector<int>count(n,0);
        for(int i=0;i<n;i++){
            count[nums[i]-1]++;
        }
        for(int i=0;i<n;i++){
            if(count[i]>1){
                return i+1;
            }
        }
        return -1;
    }
};